bindgen/private/bindgen.bzl

```diff
@@ -94,7 +94,7 @@ def rust_bindgen_library(
     rust_library(
         name = name,
         srcs = [name + "__bindgen.rs"],
-        deps = deps + [name + "__bindgen"],
```

Comment:
https://github.com/bazelbuild/rules_rust/blob/777f3e5c5d280b14d37bb7574a88f1cbfb3f8d46/test/bindgen/bindgen_test.bzl#L15-L35 

is a test where `rust_bindgen_library` is a dep of `rust_binary`. Since it works for for `rust_binary`, it's probably a bug if it doesn't work for `rusts_shared_library`.

Reply:
@daivinhtran if I'm reading the error message correctly, it seems like it's linking the cc_lib's code from the .rlib file rather than a .a file from the original cc_library target.  That would work fine if you're linking statically. 

There's a comment in _rust_bindgen_impl:
```
            # As in https://github.com/bazelbuild/rules_rust/pull/2361, we want
            # to link cc_lib to the direct parent (rlib) using `-lstatic=<cc_lib>` rustc flag
            # Hence, we do not need to provide the whole CcInfo of cc_lib because
            # it will cause the downstream binary to link the cc_lib again
            # (same effect as setting `leak_symbols` attribute above)
            # The CcInfo here only contains the custom link flags (i.e. linkopts attribute)
            # specified by users in cc_lib
```

I'm not sure if that's a good idea.  Why not let rules_cc handle the corner cases itself?   I don't have all the context here, but why can't rust_bindgen just be a code generator?  That's what I assumed it was from the documentation
> Generates a rust source file from a cc_library and a header.
hence I expected rust_bindgen_library to wrap it in a rust_library with dependencies.

Reply:
Sorry for the long-awaited reply. I was on vacation for past few weeks.

@UebelAndre

It looks like the decision of linking cc statically was originated in https://github.com/bazelbuild/rules_rust/pull/2361/files#diff-4bccb0ba6f0f44e4670d5726a5fcc04375b792fdaf7c0ad68b8891d1557c26d2R116. @neilisaac brought up a good point that this approach doesn't seem correct when in `rust_shared_library`. @UebelAndre Have you thought about this scenario before? Or was it an oversight?

Reply:
I did not think about the scenario with `rust_shared_library`. It would be very helpful if this PR could add an integration test that builds and consumes a `rust_shared_library` which uses `rust_bindgen` so we can come to an implementation that works for all use cases.